app/src/main/java/com/example/edwinmperazaduran/gridimagesearch/Dialogs/ImageFilterDialog.java

```diff
@@ -0,0 +1,155 @@
+    @Override
+    public void onAttach(Activity activity) {
+        super.onAttach(activity);
+        if (activity instanceof ImageFilterDialogListener) {
+            listener = (ImageFilterDialogListener) activity;
+        } else {
+            throw new ClassCastException(activity.toString()
+                    + " must implement MyListFragment.OnItemSelectedListener");
+        }
+    }
+
+    @Nullable
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+
+        View view = inflater.inflate(R.layout.fragment_image_filter, container);
+
+        sp_imgsize = (Spinner) view.findViewById(R.id.sp_imgsize);
+        sp_imgcolor = (Spinner) view.findViewById(R.id.sp_imgcolor);
+        sp_img_type = (Spinner) view.findViewById(R.id.sp_img_type);
+        et_img_site = (EditText) view.findViewById(R.id.et_img_site);
+        btn_save_filter = (Button) view.findViewById(R.id.btn_save_filter);
+        sp_imgsize.setFocusable(true);
+
+        if (getArguments().getSerializable("imageFilter") != null) {
+            imageFilter = (ImageFilter) getArguments().getSerializable("imageFilter");
+        }
+        //hideSoftKeyboard(view);
+        setupSpAdapter(view);
+
+        getDialog().getWindow().setSoftInputMode(
+                WindowManager.LayoutParams.SOFT_INPUT_STATE_VISIBLE);
+
+        btn_save_filter.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                hideSoftKeyboard(v);
+                finishFilter(v);
+            }
+        });
+
+        return view;
+    }
+
+    public void finishFilter(View view){
+
+        String site = et_img_site.getText().toString();
+        String size = sp_imgsize.getSelectedItem().toString();
+        String color = sp_imgcolor.getSelectedItem().toString();
+        String type = sp_img_type.getSelectedItem().toString();
+
+        imageFilter.setSize(size);
+        imageFilter.setType(type);
+        imageFilter.setColor(color);
+        imageFilter.setSite(site);
+
+        listener.onFinishDialog(imageFilter);
+        dismiss();
+    }
+
+    public void setupSpAdapter(View view){
+
+        ArrayAdapter<CharSequence> adaptercolor = ArrayAdapter.createFromResource(view.getContext(),
+                R.array.sp_img_color, android.R.layout.simple_spinner_item);
+        sp_imgcolor.setAdapter(adaptercolor);
+
+        ArrayAdapter<CharSequence> adaptersize = ArrayAdapter.createFromResource(view.getContext(),
+                R.array.sp_img_sz, android.R.layout.simple_spinner_item);
+        sp_imgsize.setAdapter(adaptersize);
+
+        ArrayAdapter<CharSequence> adaptertype = ArrayAdapter.createFromResource(view.getContext(),
+                R.array.sp_img_type, android.R.layout.simple_spinner_item);
+        sp_img_type.setAdapter(adaptertype);
+
+        if (imageFilter != null){
+            for (int i = 0; i < adaptercolor.getCount(); i++) {
```

Comment:
@edwinperaza this is pretty inefficient. The adapter already has a method to get the position of an item, you can use that to get the position and ten set the selection. ie:

```java
int post = adaptercolor.getPosition(imageFilter.getColor());
sp_imgcolor.setSelection(pos)
```